### PR TITLE
feat: signals

### LINKED ISSUE #27: Implement interface for a `signal`
## User stories
As a developer I want be able to set signals the way I want in order to control the simulation more realistically.
 
## Dependencies
Depends on: 
- #5
 - #15

## (Wiki) References
- #10
- <https://sumo.dlr.de/pydoc/traci._trafficlight.html>

## Acceptance Criteria
- [x] get/set signal state
- [ ] ~~get attached blocksections~~
- [ ] ~~get attached Fahrstrassen~~

### PR DESCRIPTION
**Pull request by SaturnHafen**
Fixes #27 
Implement signals that are linked with sumo

## PR checklist

- [x] Acceptance criteria fulfilled
- [x] Additional features are tested
- [ ] Docs (code, wiki & diagrams) updated
- [ ] ~~Breaking changes anounced~~
- [ ] Dev-branch has been merged into local branch to resolve conflicts
- [ ] Tests and linter have passed AFTER local merge
- [ ] Another dev reviewed and approved

### CODE REVIEW

File: src/wrapper/simulation_objects.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+    def __init__(self, identifier: str = None, state: "Signal.State" = State.HALT):
+        Node.__init__(self, identifier)
+        self.state = state
+
+    def update(self, data: dict):
+        return
+
+    def add_subscriptions(self) -> int:
+        return 0
```
Comment (instant-sky) [inline src/wrapper/simulation_objects.py:86]: Will these two methods be fully implemented later or do they not have to do more?

**Review by instant-sky — APPROVED** [review]
Very nice implementation. I like the documentation where necessary!

File: src/wrapper/simulation_objects.py (hunk 1 above)
Comment (SaturnHafen) [reply src/wrapper/simulation_objects.py:86]: The signal will only get updated from interlocking, so no sumo-subscription should be necessary

Comment (instant-sky) [reply src/wrapper/simulation_objects.py:86]: Ah ok, got it!

**Pull request merged**